akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -605,6 +605,14 @@ import scala.util.control.NonFatal
     // call has no effect and therefore does the right thing: nothing.
     try {
       inputs.foreach(_.onInternalError(reason))
+      // fail all sources
+      logics.foreach { logic ⇒
+        // source
+        if (logic.inCount == 0 && logic.outCount > 0) {
```

Comment:
I see. Still the complete set of things that this method does seems weird. I think the logic you try here belongs into `interpreter.finish`.